internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,265 @@
+func GetCoverage(date string, project string, level string, fuzzTarget string) (map[string]float64, map[string]float64) {
+	var fileName string
+	bucket, err := getBucket(CoverageBucket)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if level == "perProject" {
+		fileName = fmt.Sprintf("%s/reports/%s/linux/summary.json", project, date)
+	} else {
+		fileName = fmt.Sprintf("%s/fuzzer_stats/%s/%s.json", project, date, fuzzTarget)
+	}
+	reader, err := getBlob(bucket, fileName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	branchCoverage, lineCoverage, err := parseCoverage(reader)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return branchCoverage, lineCoverage
+}
+
+// GetFuzzTargets gets the list of the fuzz-targets for which fuzzing reports were generated.
+func GetFuzzTargets(project string, date string) []string {
+	bucket, err := getBucket(CoverageBucket)
+	if err != nil {
+		log.Fatal(err)
+	}
+	relativePath := fmt.Sprintf("%s/fuzzer_stats/%s", project, date)
+	blobs, err := listBlobs(bucket, relativePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fuzzTargets := make([]string, 0, len(blobs))
+	for _, blob := range blobs {
+		fuzzTargets = append(fuzzTargets, strings.Split(strings.Split(blob, "/")[3], ".")[0])
+	}
+	return fuzzTargets
+}
+
+// GetEvidences gets the list of the evidence files used by the fuzzscraper.
+func GetEvidences(project string, date string, fuzztargets []string, fuzzEngine string) map[string]string {
+	evidences := make(map[string]string)
+	evidences["revision"] = fmt.Sprintf("gs://%s/%s/srcmap/%s.json", CoverageBucket, project, date)
+	evidences["project coverage"] = fmt.Sprintf("gs://%s/%s/reports/%s/linux/summary.json", CoverageBucket, project, date)
+	for _, fuzztarget := range fuzztargets {
+		role := fmt.Sprintf("%s_%s_%v coverage", fuzzEngine, project, fuzztarget)
+		uri := fmt.Sprintf("gs://%s/%s/fuzzer_stats/%s/%v.json", CoverageBucket, project, date, fuzztarget)
+		evidences[role] = uri
+	}
+	return evidences
+}
+
+// getSingleEffort gets the fuzzingEffort from a single log file.
+func getSingleEffort(reader *storage.Reader, hash string, project string) (int, float64) {
+	var logHash string
+	var err error
+	var timeFuzz float64
+	var numTests int
+	caser := cases.Title(language.English)
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), caser.String(project)) {
+			logHash = strings.Split(scanner.Text(), " ")[1]
+		}
+		if strings.Contains(scanner.Text(), "Time ran:") {
+			timeFuzz, err = strconv.ParseFloat(strings.Split(scanner.Text(), " ")[2], 32)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		if strings.Contains(scanner.Text(), "stat::number_of_executed_units") {
+			numTests, err = strconv.Atoi(strings.Split(scanner.Text(), " ")[1])
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	if logHash == hash {
+		return numTests, timeFuzz
+	}
+	return 0, 0.0
```

Comment:
When would we return 0, 0.0? I am a bit unsure here.

Reply:
I added some comments to clarify this in the code. I return `0, 0.0` when the `commitHash` in the log file is different from the `commitHash` we are considering.